// allure-playwright/dist/helpers.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.LabelName = exports.allure = void 0;
const test_1 = __importDefault(require("@playwright/test"));
const allure_js_commons_1 = require("allure-js-commons");
const internal_1 = require("allure-js-commons/internal");
class allure {
    static addMetadataAttachment(metadata) {
        test_1.default.info().attach("allure-metadata.json", {
            contentType: internal_1.ALLURE_METADATA_CONTENT_TYPE,
            body: Buffer.from(JSON.stringify(metadata), "utf8"),
        });
    }
    static label(label, value) {
        this.addMetadataAttachment({
            labels: [{ name: label, value }],
        });
    }
    static labels(...values) {
        values.forEach(({ name, value }) => this.label(name, value));
    }
    static description(value) {
        this.addMetadataAttachment({
            description: value,
        });
    }
    static link(url, name, type) {
        this.addMetadataAttachment({
            links: [{ url, name, type }],
        });
    }
    static links(...values) {
        values.forEach(({ url, name, type }) => this.link(url, name, type));
    }
    static id(id) {
        this.label(allure_js_commons_1.LabelName.ALLURE_ID, id);
    }
    static epic(epic) {
        this.label(allure_js_commons_1.LabelName.EPIC, epic);
    }
    static feature(epic) {
        this.label(allure_js_commons_1.LabelName.FEATURE, epic);
    }
    static story(story) {
        this.label(allure_js_commons_1.LabelName.STORY, story);
    }
    static suite(name) {
        this.label(allure_js_commons_1.LabelName.SUITE, name);
    }
    static parentSuite(name) {
        this.label(allure_js_commons_1.LabelName.PARENT_SUITE, name);
    }
    static layer(layerName) {
        this.label(allure_js_commons_1.LabelName.LAYER, layerName);
    }
    static subSuite(name) {
        this.label(allure_js_commons_1.LabelName.SUB_SUITE, name);
    }
    static owner(owner) {
        this.label(allure_js_commons_1.LabelName.OWNER, owner);
    }
    static severity(severity) {
        this.label(allure_js_commons_1.LabelName.SEVERITY, severity);
    }
    static tag(tag) {
        this.label(allure_js_commons_1.LabelName.TAG, tag);
    }
    static tags(...values) {
        values.forEach((value) => this.tag(value));
    }
    static issue(name, url) {
        this.link(url, name, allure_js_commons_1.LinkType.ISSUE);
    }
    static tms(name, url) {
        this.link(url, name, allure_js_commons_1.LinkType.TMS);
    }
    static parameter(name, value, options) {
        this.addMetadataAttachment({
            parameter: [
                Object.assign({ name,
                    value }, options),
            ],
        });
    }
    static addParameter(name, value, options) {
        this.parameter(name, value, options);
    }
}
exports.allure = allure;
var allure_js_commons_2 = require("allure-js-commons");
Object.defineProperty(exports, "LabelName", { enumerable: true, get: function () { return allure_js_commons_2.LabelName; } });
